hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/connector/mongodb/MongoTable.java

```diff
@@ -0,0 +1,180 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://hazelcast.com/hazelcast-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.hazelcast.jet.sql.impl.connector.mongodb;
+
+
+import com.hazelcast.function.SupplierEx;
+import com.hazelcast.jet.sql.impl.connector.SqlConnector;
+import com.hazelcast.jet.sql.impl.schema.JetTable;
+import com.hazelcast.sql.impl.extract.QueryTarget;
+import com.hazelcast.sql.impl.optimizer.PlanObjectKey;
+import com.hazelcast.sql.impl.schema.TableField;
+import com.hazelcast.sql.impl.schema.TableStatistics;
+import com.hazelcast.sql.impl.type.QueryDataType;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+
+import static java.util.stream.Collectors.toList;
+
+class MongoTable extends JetTable {
+
+    final String databaseName;
+    final String collectionName;
+    final String connectionString;
+    final Map<String, String> options;
+    /**
+     * Streaming query always needs _id to be present, even if user don't request it
+     */
+    final boolean streaming;
+
+    MongoTable(
+            @Nonnull String schemaName,
+            @Nonnull String name,
+            @Nonnull String databaseName,
+            @Nullable String collectionName,
+            @Nonnull Map<String, String> options,
+            @Nonnull SqlConnector sqlConnector,
+            @Nonnull List<TableField> fields,
+            @Nonnull TableStatistics statistics, boolean streaming) {
+        super(sqlConnector, fields, schemaName, name, statistics);
+        this.databaseName = databaseName;
+        this.collectionName = collectionName;
+        this.options = options;
+        this.connectionString = options.get(Options.CONNECTION_STRING_OPTION);
+        this.streaming = streaming;
+    }
+
+    public MongoTableField getField(String name) {
+        List<TableField> fields = getFields();
+        for (TableField field : fields) {
+            if (field.getName().equals(name)) {
+                return (MongoTableField) field;
+            }
+        }
+        throw new IllegalArgumentException("field " + name + " does not exist");
+    }
+    public MongoTableField getFieldByExternalName(String name) {
+        return getFields().stream()
+                          .map(f -> (MongoTableField) f)
+                          .filter(f -> f.getExternalName().equals(name))
+                          .findFirst()
+                          .orElseThrow(() -> new IllegalArgumentException("field " + name + " does not exist"));
+    }
+
+    String[] paths() {
```

Comment:
nit : Save the result of paths() in a field ? Instead of calculating each time. The same goeas for types() etc.